### ISSUE TITLE
Please ungrade the vertx dependency to latest one

### ISSUE THREAD

**Issue by mxmind** [weld/weld-vertx#39]
The latest vert.x version is 3.7.0, but yours is 3.5.4.

**Comment by manovotn** [comment]
@mkouba would it make sense to have the vertx dependency `provided`?
Since we are not making many changes in this project, we would fall behind in versions easily.

**Comment by manovotn** [comment]
Oop, accidental close instead of comment.
I meant to say we totally should and perhaps cut a release with it.
@mkouba would it make sense to have vertx as provided dependency here? We don't make many changes to the project and hence we easily miss newer versions. OTOH it would be a breaking change.

**Comment by mkouba** [comment]
> would it make sense to have vertx as provided dependency here?

I don't think this would help much. Even now you can specify a vertx dependency on the same level as the weld-vertx dependency and the version used by weld should be overriden.

@mxmind Feel free to send a PR ;-). 

Also note that most of the weld-vertx functionalities were ported to https://github.com/quarkusio/quarkus. And it's very likely that the weld-vertx project will be switched to the "maintenance mode".

**Comment by mxmind** [comment]
Hi @mkouba, I'll prepare PR in a few days.
I'm looking on quarkus with very huge interest, but the project I'm working in is too big and, at least in the nearest future, the migration on quarkus will be very problematic for us.

**Comment by mkouba** [comment]
Thanks @mxmind. Just out of curiosity - which of the weld-vertx features are you using in your project?

**Comment by mxmind** [comment]
Hi @mkouba, 
Sorry for a late response.
We heavily depend on AsyncWorkers, AsyncReferences, and Service Discovery features. We have own mechanism to work with local and clustered event buses, but it based on your CDI Vertx extension.
Another thing, which I still cannot resolve using `quarkus`, I don't know how to properly scale the number of service verticles.
